src/encoded.ts

```diff
@@ -0,0 +1,35 @@
+import {Checksum256, Name, Struct} from '@wharfkit/antelope'
+import {RestoreArgs, SerializedSession, Session} from './index-module'
+
+/**
+ * The metadata of an [[AccountCreationPlugin]].
+ */
+@Struct.type('url_encoded_session')
+export class URLEncodedSession extends Struct {
+    @Struct.field(Checksum256) declare chain: Checksum256
+    @Struct.field(Name) declare actor: Name
+    @Struct.field(Name) declare permission: Name
+    @Struct.field('string') declare walletPlugin: string
+    @Struct.field('string', {optional: true}) declare data?: string
+
+    static fromSession(data: Session | SerializedSession): URLEncodedSession {
```

Comment:
Do we need some better error handling of bad data being passed in this method?

Reply:
The serializer should catch any bad data and report on it.